src/hooks/useCookieState.ts

```diff
@@ -0,0 +1,31 @@
+import Cookies from 'js-cookie';
+import { useCallback, useState } from 'react';
+
+const useCookieState = (
+  cookieName: string
+): [
+  string | null,
+  (newValue: string, options?: Cookies.CookieAttributes) => void,
+  () => void
+] => {
+  const [value, setValue] = useState<string | null>(
+    () => Cookies.get(cookieName) || null
```

Comment:
`Get` should also use the same options that were used in `set`

Reply:
Als this breaks on the server side. So on the server side we need to return null if window is not defined.